src/v2/view-builder/internals/BaseOktaVerifyChallengeView.js

```diff
@@ -108,6 +108,10 @@ const Body = BaseFormWithPolling.extend({
               foundPort = true;
               if (deviceChallenge.enhancedPollingEnabled !== false) {
                 this.stopPolling();
+                this.trigger('save', this.model);
+                // if this poll doesn't return a different remediation, start polling again
+                this.startPolling();
```

Comment:
is the purpose of stopPolling->startPolling to avoid two poll at about the same time?

Reply:
@yannongli-okta can you confirm?

Reply:
Also, do we know if `this.trigger('save', this.model)` is synchronous? I'm guessing not.

I'm asking because we cannot assume that just because `this.trigger('save', this.model)` has executed that the response has already been returned and processed, so perhaps restarting polling is not appropriate yet. A simple example would be if the network speed is slow and the response just hadn't been returned yet, we would restart polling prematurely.

Reply:
@yannongli-okta could this cause an increase in the number of "improper poll" requests if the response is very slow, which may cause users to see _more_ errors in the widget? Before this change, we would simply wait for a slow response to complete, but now we start the polling timer again.

Reply:
@shawyu-okta the issue is, before this change it would only do one polling which is triggered by this.trigger('save', this.model);. if it's returning the same remediation as before (device-challenge-poll), UI will not refresh and there will be no further polling, hence user is stuck there. this is a worse case in general